components/sidebar/sidebar_channel/sidebar_channel_menu/sidebar_channel_menu.test.tsx

```diff
@@ -2,238 +2,42 @@
+    const useDispatchMock = jest.spyOn(reactRedux, 'useDispatch');
+    const useSelectorMock = jest.spyOn(reactRedux, 'useSelector');
+    const mockStore = configureStore();
+
+    beforeEach(() => {
+        useDispatchMock.mockClear();
+        useSelectorMock.mockClear();
+    });
+
     const testChannel = TestHelper.getChannelMock();
-    const testCategory = TestHelper.getCategoryMock();
 
     const baseProps = {
         channel: testChannel,
         channelLink: 'http://a.fake.link',
-        categories: [testCategory],
-        currentUserId: 'user_id',
-        currentCategory: testCategory,
-        currentTeamId: 'team_id',
-        isUnread: false,
-        isFavorite: false,
-        isMuted: false,
-        managePublicChannelMembers: true,
-        managePrivateChannelMembers: true,
-        closeHandler: jest.fn(),
-        isCollapsed: false,
-        isMenuOpen: true,
-        onToggleMenu: jest.fn(),
-        multiSelectedChannelIds: [],
-        displayedChannels: [],
-        actions: {
-            markChannelAsRead: jest.fn(),
-            favoriteChannel: jest.fn(),
-            unfavoriteChannel: jest.fn(),
-            muteChannel: jest.fn(),
-            unmuteChannel: jest.fn(),
-            openModal: jest.fn(),
-            createCategory: jest.fn(),
-            addChannelsInSidebar: jest.fn(),
-        },
+        location: 'sidebar',
     };
 
     test('should match snapshot and contain correct buttons', () => {
-        const wrapper = shallowWithIntl(
-            <SidebarChannelMenu {...baseProps}/>,
-        );
-
-        expect(wrapper.find('#favorite-channel_id')).toHaveLength(1);
-        expect(wrapper.find('#mute-channel_id')).toHaveLength(1);
-        expect(wrapper.find('#copyLink-channel_id')).toHaveLength(1);
-        expect(wrapper.find('#addMembers-channel_id')).toHaveLength(1);
-        expect(wrapper.find('#leave-channel_id')).toHaveLength(1);
-
-        expect(wrapper).toMatchSnapshot();
-    });
-
-    test('should match snapshot when channel is unread', () => {
-        const props = {
-            ...baseProps,
-            isUnread: true,
-        };
-
-        const wrapper = shallowWithIntl(
-            <SidebarChannelMenu {...props}/>,
-        );
-
-        expect(wrapper.find('#markAsRead-channel_id')).toHaveLength(1);
-
-        expect(wrapper).toMatchSnapshot();
-    });
-
-    test('should match snapshot when channel is favorite', () => {
-        const props = {
-            ...baseProps,
-            isFavorite: true,
+        const state = {
         };
+        const store = mockStore(state);
+        const dummyDispatch = jest.fn();
+        useDispatchMock.mockReturnValue(dummyDispatch);
 
-        const wrapper = shallowWithIntl(
-            <SidebarChannelMenu {...props}/>,
+        const wrapper = shallow(
+            <reactRedux.Provider store={store}><SidebarChannelMenu {...baseProps}/></reactRedux.Provider>,
```

Comment:
This is not the final test it was just a quick fix so that nothing fails, i have the ticket to write proper tests here https://mattermost.atlassian.net/browse/MM-44901

Didn't want this to be too cluttered :P